meson.build

```diff
@@ -0,0 +1,150 @@
+project('GMime',
+	['c'],
+	license: 'GPLv2',
+    version: '3.2.13',
+    meson_version: '>=1.0.0',
+    default_options : [
+    ])
+
+gmime_version = meson.project_version()
+ver_arr = gmime_version.split('.')
+
+GMIME_MAJOR_VERSION=ver_arr[0].to_int()
+GMIME_MINOR_VERSION=ver_arr[1].to_int()
+GMIME_MICRO_VERSION=ver_arr[2].to_int()
+GMIME_BINARY_AGE= 100 * GMIME_MINOR_VERSION + GMIME_MICRO_VERSION
+GMIME_INTERFACE_AGE=0
+
+gmime_package = 'gmime-@0@.0'.format(GMIME_MAJOR_VERSION)
+
+add_global_arguments('-DHAVE_CONFIG_H=1', language : 'c')
+
+iconv = dependency('iconv', required : false, method: get_option('iconv'))
```

Comment:
By the way, marking it as `required: false` seems wrong. The configure.ac considers it mandatory, and the source code unconditionally tries to use it. All you're doing here is adding it to the list of deps -- you don't check whether it's available and set different preprocessor defines to e.g. avoid using it if it doesn't exist.

Reply:
Yeah, required: false is wrong. It was something left over. Removed it.

The reason for the get_option is to be able to specify what version of iconv to use if you have multiple installed. This is something configure.ac included. Maybe it's considered bad practice or there is a better way to do it in now?

Reply:
I have no idea what the original intention was, but the general rule of thumb is that you can either get the standalone gnu one (always an external library) or rely on the platform one (which may be a builtin or an external library, and for glibc it's also the same underlying codebase as the gnu one).

Which one you use isn't much of an option, though, because they both require the same `iconv.h` header and if you have both available then the gnu one overrides the header search path. It also overrides the symbol names, so if you don't perform a link verification test using the header (the dependency() provided by meson does this for you) you may end up compiling against the default header which is the gnu one, but linking to an incompatible BSD or musl one that is built into your libc.

The only choice you have really is to either:
- use the one picked up by auto
- return an error or simply disable iconv support if you didn't get the one you wanted
- modify CFLAGS and LDFLAGS *before* running meson, to prioritize a custom copy in the location you prefer


You're effectively offering an opt-in for option 2 (return a fatal error). But I traditionally wonder: does it actually matter which iconv you find?

Reply:
> I have no idea what the original intention was, but the general rule of thumb is that you can either get the standalone gnu one (always an external library) or rely on the platform one (which may be a builtin or an external library, and for glibc it's also the same underlying codebase as the gnu one).

This was the original intention, yes.